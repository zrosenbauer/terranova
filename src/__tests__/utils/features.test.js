import {
  destroyTerra,
  createTerra,
  getDataById,
} from '../../utils/terra';
import {
  getFeatureBounds,
  getFeatureByMapId,
  getFeatureProperties,
  getFeatureState,
  getFeatureType,
  setFeatureState,
  setFeatureType,
} from '../../utils/features';

describe('utils/features', () => {
  const id = 'terra-id';

  beforeEach(() => {
    document.body.innerHTML = `<div id="${id}" />`;
    createTerra(id);
  });

  afterEach(() => {
    destroyTerra(id)
  });

  describe('getFeatureByMapId', () => {
    it('returns null if not found', () => {
      expect(getFeatureByMapId('not-found')).toBe(null);
    });

    it('returns feature if found', () => {
      const featureId = 'feature-id';
      expect(getFeatureByMapId(id, featureId)).toBe(featureId);
    });
  });
});
