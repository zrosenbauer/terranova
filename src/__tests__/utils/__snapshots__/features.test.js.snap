// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/features matches snap shot for events constant 1`] = `
Object {
  "click": "click",
  "mouseOut": "mouseout",
  "mouseOver": "mouseover",
}
`;
